From d079631ab3a87d13fb896bc398879ace4603ad24 Mon Sep 17 00:00:00 2001
From: John Crispin <john@openwrt.org>
Date: Thu, 16 Mar 2023 20:11:46 +0530
Subject: [PATCH 169/500] OpenWrt: 920-device_tree_cmdline.patch

of/ftd: add device tree cmdline

Signed-off-by: John Crispin <john@openwrt.org>

Change-Id: Ib9b64b4562173180ea9c995f197028c937249574
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
Signed-off-by: Saravanan Jaganathan <quic_sjaganat@quicinc.com>
(cherry picked from commit f0aed29e6bf5123c2fc784e3ff77c4814d464fd3)
---
 drivers/of/fdt.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/of/fdt.c b/drivers/of/fdt.c
index bf502ba8da95..1886102b2382 100644
--- a/drivers/of/fdt.c
+++ b/drivers/of/fdt.c
@@ -1185,6 +1185,9 @@ int __init early_init_dt_scan_chosen(char *cmdline)
 	p = of_get_flat_dt_prop(node, "bootargs", &l);
 	if (p != NULL && l > 0)
 		strscpy(cmdline, p, min(l, COMMAND_LINE_SIZE));
+	p = of_get_flat_dt_prop(node, "bootargs-append", &l);
+	if (p != NULL && l > 0)
+		strlcat(cmdline, p, min_t(int, strlen(cmdline) + (int)l, COMMAND_LINE_SIZE));
 
 handle_cmdline:
 	/*
-- 
2.34.1

